File: c/alternate_case.c
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int alternateCase(const char *input, char *output, size_t output_size) {
    int makeUpper = 1;
    size_t i;
    for (i = 0; input[i] != '\0'; i++) {
        if (i + 1 >= output_size) {
            return -1; /* output buffer too small */
        }
        if (isalpha((unsigned char)input[i])) {
            output[i] = makeUpper ? toupper((unsigned char)input[i])
                                  : tolower((unsigned char)input[i]);
            makeUpper = !makeUpper;
        } else {
            output[i] = input[i];
        }
    }
    output[i] = '\0';
    return 0;
}

int main(void) {
    char output[256];

    alternateCase("instagram", output, sizeof(output));
    printf("%s\n", output);  /* InStAgRaM */

    alternateCase("hello world", output, sizeof(output));
    printf("%s\n", output);  /* HeLlO wOrLd */

    alternateCase("abc123def", output, sizeof(output));
    printf("%s\n", output);  /* AbC123DeF */

    return 0;
}

File: python/alternate_case.py
def alternate_case(text: str) -> str:
    make_upper = True
    result = []
    for ch in text:
        if ch.isalpha():
            result.append(ch.upper() if make_upper else ch.lower())
            make_upper = not make_upper
        else:
            result.append(ch)
    return "".join(result)


if __name__ == "__main__":
    print(alternate_case("instagram"))   # InStAgRaM
    print(alternate_case("hello world")) # HeLlO wOrLd
    print(alternate_case("abc123def"))   # AbC123DeF
